pkg/cli/mirror/fbc_operators.go

```diff
@@ -537,89 +546,52 @@ func findFirstAvailableMirror(ctx context.Context, mirrors []sysregistriesv2.End
 	return "", finalError
 }
 
-// getManifest reads the manifest of the OCI FBC image
+// getManifest reads the manifest of the OCI FBC image. If imgSrc is a manifest
+// list, it will return the first manifest available in the list
 // and returns it as a go structure of type manifest.Manifest
 func getManifest(ctx context.Context, imgSrc types.ImageSource) (manifest.Manifest, error) {
```

Comment:
getManifest doesn't seem to be called anymore. Did I miss something?

Reply:
Since the purpose of `findFirstAvailableMirror` is to simply "poke" at the provided image reference to see if it exists (by attempting to get the image source and fetching its manifest and simply checking err == nil), it does not matter which manifest you get back from the `getManifest` function. In other words, this call:

```go
_, _, err = regFuncs.getManifest(ctx, nil, imgsrc)
```

takes an types.ImageSource and calls `types.ImageSource.GetManifest(ctx, nil)`. For a docker image source it's going to fetch the manifest for the tag/sha stored in the image source (which could refer to an image or index). For an oci image source assuming that the reference is `oci:<path>` (and not `oci:<path>:<image>`) and there's only one item in the index.json, you get that manifest (which could refer to an image or index). Bear in mind that the way that  `findFirstAvailableMirror` is called, we're only dealing with docker image references, so the comments about OCI are moot.

Bottom line is that I did know this `getManifest` function was no longer used, but I left this in place since it could be useful. If you'd like I can simply remove it and its associated test case since it's not used anywhere at this point.

Reply:
You're right for the `types.ImageSource.GetManifest(ctx, nil)`
For the local `getManifest`, removing would help eliminate confusion about things. WDYT?
We could still find it again in repo history.

Reply:
Agreed with removing the dead code. Changed in commit https://github.com/openshift/oc-mirror/pull/611/commits/e371440b76d901fc44bbf7b05929321024b28ea0